lightning/src/routing/scoring.rs

```diff
@@ -648,18 +655,18 @@ impl<T: Time> ChannelLiquidity<T> {
 impl<L: Deref<Target = u64>, T: Time, U: Deref<Target = T>> DirectedChannelLiquidity<L, T, U> {
 	/// Returns the success probability of routing the given HTLC `amount_msat` through the channel
 	/// in this direction.
-	fn success_probability(&self, amount_msat: u64) -> f64 {
+	fn success_probability(&self, amount_msat: u64) -> Probability {
 		let max_liquidity_msat = self.max_liquidity_msat();
 		let min_liquidity_msat = core::cmp::min(self.min_liquidity_msat(), max_liquidity_msat);
 		if amount_msat > max_liquidity_msat {
-			0.0
+			Probability::Zero
```

Comment:
Why not just do the log here instead of returning the enum? Feels like kinda useless branches, no?